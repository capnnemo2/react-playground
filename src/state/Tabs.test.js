import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Tabs from './Tabs';

describe('Tabs Component', () => {
    const tabsProps = [
        { name: 'First tab',
          content: 'This is the content of the first tab.'},
        { name: 'Second tab',
          content: 'This is the content of the second tab. There is more writing here than in the first tab.'},
        { name: 'Third tab',
          content: 'This is the content of the third tab. There is more writing her than in the first tab. There is also more writing here than in the second tab.'}
      ]

    it('renders without errors', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Tabs />, div);
        ReactDOM.unmountComponentAtNode(div);
    })

    it('renders the first tab by default', () => {
        const wrapper = shallow(<Tabs tabs={tabsProps} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('renders empty given no tabs', () => {
        const wrapper = shallow(<Tabs />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('closees the first tab and opens any clicked tab', () => {
        const wrapper = shallow(<Tabs tabs={tabsProps} />)
        wrapper.find('button').at(1).simulate('click')
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})