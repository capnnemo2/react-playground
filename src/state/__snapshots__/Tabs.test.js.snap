// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs Component closees the first tab and opens any clicked tab 1`] = `
<div>
  <br />
  <br />
  <h2>
    Tabs project
  </h2>
  <button
    key="0"
    onClick={[Function]}
  >
    First tab
  </button>
  <button
    key="1"
    onClick={[Function]}
  >
    Second tab
  </button>
  <button
    key="2"
    onClick={[Function]}
  >
    Third tab
  </button>
  <div
    className="content"
  >
    This is the content of the second tab. There is more writing here than in the first tab.
  </div>
</div>
`;

exports[`Tabs Component renders empty given no tabs 1`] = `
<div>
  <br />
  <br />
  <h2>
    Tabs project
  </h2>
</div>
`;

exports[`Tabs Component renders the first tab by default 1`] = `
<div>
  <br />
  <br />
  <h2>
    Tabs project
  </h2>
  <button
    key="0"
    onClick={[Function]}
  >
    First tab
  </button>
  <button
    key="1"
    onClick={[Function]}
  >
    Second tab
  </button>
  <button
    key="2"
    onClick={[Function]}
  >
    Third tab
  </button>
  <div
    className="content"
  >
    This is the content of the first tab.
  </div>
</div>
`;
